common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
We should introduce the plugin model - to give the option of opting out of/into running the tracing for example; for several reasons: unknown overhead in context, debugging/development vs production, redundancy with other models that may already be in use, ... The tracing information has to round trip through Kafka; what's the cost of doing that? Once we have the benchmarking strategy established in the open it makes it easier to then speak about the effect of adding tracing on the critical path in this way. 

I know this information is wholly useful for operators and at the same I see how we can even make it visible to end users; it would be very powerful to that. Further I rather we don't introduce new technical debt when we know it's debt out the gate. I think this feature can continue to be developed but doesn't that's to impact the core with the right architecture: the plugin mode is one suggestion for how to do that.